src/backend/metal/src/command.rs

```diff
@@ -599,6 +599,17 @@ impl RawCommandBuffer<Backend> for CommandBuffer {
                                         }
                                     }
                                 }
+                                ConstantBuffer(ref buffers) => {
+                                    if let EncoderState::Render(ref encoder) = inner.encoder_state {
```

Comment:
looks like we'll miss a call to `add_buffers` if the encoder is not recording. We should call it before checking the encoder

Reply:
Ah so thats why it was done like that above, I thought it wouldn't matter then.

Reply:
I haven't double-checked if it matters, tbh, assumed it does